File: grupoDeSlides_01-02/ex-03/ex-03_java/ex-03/src/main/java/Caixa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author Robson
 */
public class Caixa {
    
    //atributos da caixa
    private double largura;
    private double altura;
    private double profundidade;
    
    public boolean setLargura(double l)
    {
        if (l > 0)
        {
            largura = l;
            return true;
        }
        return false;
    }
    
    public boolean setAltura(double a)
    {
        if (a > 0)
        {
            altura = a;
            return true;
        }
        return false;
    }
    public boolean setProfundidade(double p)
    {
        if (p > 0)
        {
            profundidade = p;
            return true;
        }
        return false;
    }
    
    public double calcularAreaExt()
    {
        return 2*(altura*largura + 
                altura*profundidade +
                largura*profundidade);
    }
    
    public double calcularVolume()
    {
        return profundidade*largura*altura;
    }
    
}

File: ex-03/caixa.cpp
#include "caixa.hpp"

Caixa::Caixa() : mLargura{0}, mAltura{0}, mProfundidade{0} 
{

}

Caixa::~Caixa() 
{

}

bool Caixa::SetLargura(int largura)
{
  if(largura <= 0) 
  {
      return false;
  }
  mLargura = largura;
  return true;
}

bool Caixa::SetAltura(int altura)
{
  if(altura <= 0)
  {
      return false;
  }
  mAltura = altura;
  return true;
}

bool Caixa::SetProfundidade(int profundidade)
{
  if(profundidade <= 0)
  {
      return false;
  }
  mProfundidade = profundidade;
  return true;
}

int Caixa::CalcAreaExt()
{
  return 2 * (mLargura * mAltura + mLargura * mProfundidade + mAltura * mProfundidade);
}

int Caixa::CalcVolume()
{
  return mLargura * mAltura * mProfundidade;
}
